virtio-devices/src/vhost_user/blk.rs

```diff
@@ -37,18 +39,32 @@ impl VhostUserMasterReqHandler for SlaveReqHandler {}
 pub struct Blk {
     common: VirtioCommon,
     id: String,
-    vhost_user_blk: Master,
+    vhost_user_blk: Option<Master>,
     config: VirtioBlockConfig,
     seccomp_action: SeccompAction,
     guest_memory: Option<GuestMemoryAtomic<GuestMemoryMmap>>,
     acked_protocol_features: u64,
+    disconnected: Arc<AtomicBool>,
+    sock_path: String,
+    activate_evt: EventFd,
+    first_activated: AtomicBool,
 }
 
 impl Blk {
     /// Create a new vhost-user-blk device
-    pub fn new(id: String, vu_cfg: VhostUserConfig, seccomp_action: SeccompAction) -> Result<Blk> {
-        let mut vhost_user_blk = Master::connect(&vu_cfg.socket, vu_cfg.num_queues as u64)
-            .map_err(Error::VhostUserCreateMaster)?;
+    pub fn new(
+        id: String,
+        vu_cfg: VhostUserConfig,
+        seccomp_action: SeccompAction,
+        activate_evt: EventFd,
+    ) -> Result<Blk> {
+        let mut vhost_user_blk = retry_with_index(delay::Fixed::from_millis(500), |retry_times| {
+            if retry_times == 1 {
+                println!("waiting for vhost-user-blk deamon opening...");
+            }
+            Master::connect(&vu_cfg.socket, vu_cfg.num_queues as u64)
+        })
+        .unwrap();
```

Comment:
Is there a way to return a sensible error? Or are we better off panicking the whole program if vhost-user-blk is not reachable? What is your thought here?

Reply:
I replaced `unwrap()` by a `map_err()?`.